### ISSUE TITLE
Add cron service

### ISSUE THREAD

**Issue by shieldwed** [nextcloud/docker#36]
In order to keep things easy for image users (less additional config required), add cron service similar to wonderfall/nextcloud. This might require a service manager like s6. Nexcloud can be preconfigured to use cron automatically.

**Comment by SnowMB** [comment]
I fiddled around a while ago and found a solution with cron and supervisord. 

https://github.com/SnowMB/docker/tree/adding_cron

Would that be sufficient?

**Comment by shieldwed** [comment]
Yes, certainly it gets the job done. I was thinking of doing something similar. You were just a tiny bit quicker.

**Comment by shieldwed** [comment]
Are you going to create a pull request?

**Comment by pierreozoux** [comment]
The philosophy of Docker is to have one process per container.
So we shouldn't use supervisord for instance.
The best is then to use a side process for this kind of tasks.
I agree it is not the simplest for the beginner end user.

As an official image, it is not possible to use supervisord.
This is a nice building block to build more complex setup.
In the context of kubernetes for instance, the cron tasks are native from the cluster solution.
It allows fine grain permission/resource allocation.
Once we have kubernetes support, it doesn't make sense to ship cron for kubernetes users.
For now, there is docker-compose support from myself and @SnowMB and it ships with cron.

I'm closing in favor of #37

**Comment by shieldwed** [comment]
Thank you for pointing out, that there is no need to have cron within the image. You can use the external cron and docker exec just as well.

**Comment by Richie765** [comment]
You can also put something like this in your compose-file:

```yaml
  cron:
    image: nextcloud:14.0.3
    restart: always
    volumes:
      - data:/var/www/html
    entrypoint: |
      bash -c 'bash -s <<EOF
      trap "break;exit" SIGHUP SIGINT SIGTERM
      while /bin/true; do
        su -s "/bin/bash" -c "/usr/local/bin/php /var/www/html/cron.php" www-data
        echo $$(date) - Running cron finished
        sleep 900
      done
      EOF'
```

**Comment by odalle** [comment]
> You can also put something like this in your compose-file:
> 
> ```yaml
>   cron:
>     image: nextcloud:14.0.3
>     restart: always
>     volumes:
>       - data:/var/www/html
>     entrypoint: |
>       bash -c 'bash -s <<EOF
>       trap "break;exit" SIGHUP SIGINT SIGTERM
>       while /bin/true; do
>         su -s "/bin/bash" -c "/usr/local/bin/php /var/www/html/cron.php" www-data
>         echo $$(date) - Running cron finished
>         sleep 900
>       done
>       EOF'
> ```

It seems simple and easy. Thanks! 

With example [docker-compose/with-nginx-proxy/mariadb/fpm/](https://github.com/nextcloud/docker/tree/master/.examples/docker-compose/with-nginx-proxy/mariadb/fpm), the `docker-compose.yml` just needs to be changed as follows to share the right volume and use same image:

```yaml
cron:
    image: nextcloud:fpm
    restart: always
    volumes:
      - nextcloud:/var/www/html
    depends_on:
      - app
    entrypoint: |
      bash -c 'bash -s <<EOF
      trap "break;exit" SIGHUP SIGINT SIGTERM
      while /bin/true; do
        su -s "/bin/bash" -c "/usr/local/bin/php /var/www/html/cron.php" www-data
        echo $$(date) - Running cron finished
        sleep 900
      done
      EOF'
```

**Comment by snw35** [comment]
For those using the alpine-based fpm container, this works well:
```
  nc-cron:
      image: nextcloud:fpm-alpine
      restart: always
      volumes:
        - nc-dir:/var/www/html
      depends_on:
        - nc-fpm
      networks:
        - nc-backend
      entrypoint: |
        sh -c 'sh -s <<EOF
        trap "break;exit" SIGHUP SIGINT SIGTERM
        while /bin/true; do
          su -s "/bin/sh" -c "/usr/local/bin/php /var/www/html/cron.php" www-data
          echo $$(date) - Running cron finished
          sleep 900
        done
        EOF'
```

**Comment by chrstn-hntschl** [comment]
While this works for me, the `$$(date) ` command seems to be evaluated only once so the log contains the same timestamp endlessly, which makes debugging a bit difficult. This is a bash question - so any idea how to improve on this?

**Comment by mintalicious** [comment]
There is a cron.sh script inside the nextcloud image which can be used. Have a look at the cron container inside https://github.com/nextcloud/docker/blob/master/.examples/docker-compose/with-nginx-proxy/mariadb-cron-redis/apache/docker-compose.yml

**Comment by Chartman123** [comment]
> There is a cron.sh script inside the nextcloud image which can be used. Have a look at the cron container inside https://github.com/nextcloud/docker/blob/master/.examples/docker-compose/with-nginx-proxy/mariadb-cron-redis/apache/docker-compose.yml
> 
> ```yaml
>   cron:
>     image: nextcloud:17
>     restart: always
>     volumes:
>       - nextcloud:/var/www/html
>     entrypoint: /cron.sh
>     depends_on:
>       - db
> ```

I just tried this in my docker-compose, but Nextcloud tells me on the cron status page that the cron job isn't running. Is this normal? I can see the cron job running in the logs of the cron container.

**Comment by mintalicious** [comment]
> > There is a cron.sh script inside the nextcloud image which can be used. Have a look at the cron container inside https://github.com/nextcloud/docker/blob/master/.examples/docker-compose/with-nginx-proxy/mariadb-cron-redis/apache/docker-compose.yml
> > ```yaml
> >   cron:
> >     image: nextcloud:17
> >     restart: always
> >     volumes:
> >       - nextcloud:/var/www/html
> >     entrypoint: /cron.sh
> >     depends_on:
> >       - db
> > ```
> 
> I just tried this in my docker-compose, but Nextcloud tells me on the cron status page that the cron job isn't running. Is this normal? I can see the cron job running in the logs of the cron container.

Hi, this is what's working for me (my complete docker-compose.yml):
```yaml
version: '3' 

services:
  nextcloud:
    image: nextcloud:18
    restart: unless-stopped
    volumes:
      - ./nextcloud_data:/var/www/html
    links:
      - mariadb
    environment:
      - VIRTUAL_HOST=nextcloud.example.com
      - LETSENCRYPT_HOST=nextcloud.example.com
    networks:
      - proxynet
      - backend

  cron:
    image: nextcloud:18
    restart: unless-stopped
    volumes:
      - ./nextcloud_data:/var/www/html
    networks:
      - backend
    entrypoint: /cron.sh

  mariadb:
    image: mariadb
    restart: unless-stopped
    environment:
      - MYSQL_ROOT_PASSWORD=secret
    volumes:
      - ./mariadb_data:/var/lib/mysql
      - ./mariadb_config:/etc/mysql/conf.d
    networks:
      - backend

networks:
  backend:
  proxynet:
    external:
      name: webproxy
```

**Comment by Chartman123** [comment]
Thank you very much, my compose file was missing the nextcloud network :+1:
